zanata-war/src/main/java/org/zanata/action/VersionGroupJoinAction.java

```diff
@@ -149,14 +152,47 @@ public String send() {
         }
     }
 
-    @AllArgsConstructor
-    public final class SelectableProject {
+    /**
+     * This is to reset data when user closes dialog or after sending email.
+     * See version-group/request_join_modal.xhtml#cancelJoinGroupEmail
+     */
+    public void clearFormFields() {
+        projectSlug = "";
+        projectVersions.clear();
+        setQuery("");
+    }
 
-        @Getter
-        private HProjectIteration projectIteration;
+    // clear all the project versions in the selected project
+    private void clearProjectVersions() {
+        if(!projectVersions.isEmpty()) {
+            projectVersions.clear();
```

Comment:
- [ ] Is there any point to this? `.clear()` should work the same regardless whether the list is empty, so it looks like this is just extra code to maintain.